dynamiqs/solvers/utils/batching.py

```diff
@@ -5,16 +5,36 @@
 
 def batch_H(H: TDTensor) -> TDTensor:
     # handle H batching
-    # x: (b_H?, n, n) ->  (b_H, 1, n, n)
+    # x: (b_H?, n, n) ->  (b_H, 1, 1, n, n)
     if H.dim() == 2:  # (n, n)
         H = H.unsqueeze(0)  # (1, n, n)
```

Comment:
```suggestion
    if H.dim() == 2:  # (n, n)
        return H[None, None, None, ...]
    elif H.dim() == 3: # (b_H, n, n)
        return H[:, None, None, ...]
    else:
        raise ValueError(f"Expected `H` of dimension 2 or dimension 3 if batched, got {H.ndim} dimensions with shape {H.shape}.")
```

Reply:
IMO it makes it more explicit how we're batching instead of using repeated `.unsqueeze` calls. Also added a check for the dimension. Could you do the same for `batch_y0` and `batch_jump_ops` ?

Reply:
I cannot do it on `H` because it's not a `Tensor` but a `TDTensor` (which only has `unsqueeze` and does not support slicing). It's cleaner for L and rho0 so I'll do it

Reply:
Ah right I forgot !